packages/zipkin-instrumentation-request-promise/test/integrationTest.js

```diff
@@ -0,0 +1,275 @@
+          const initialTraceId = annotations[0].traceId.traceId;
+          annotations.forEach(ann => expect(ann.traceId.traceId)
+            .to.equal(initialTraceId).and
+            .to.have.lengthOf(16));
+
+          expect(annotations[0].annotation.annotationType).to.equal('ServiceName');
+          expect(annotations[0].annotation.serviceName).to.equal('weather-app');
+
+          expect(annotations[1].annotation.annotationType).to.equal('Rpc');
+          expect(annotations[1].annotation.name).to.equal('GET');
+
+          expect(annotations[2].annotation.annotationType).to.equal('BinaryAnnotation');
+          expect(annotations[2].annotation.key).to.equal('http.url');
+          expect(annotations[2].annotation.value).to.equal(url);
+
+          expect(annotations[3].annotation.annotationType).to.equal('ClientSend');
+
+          expect(annotations[4].annotation.annotationType).to.equal('ServerAddr');
+
+          expect(annotations[5].annotation.annotationType).to.equal('BinaryAnnotation');
+          expect(annotations[5].annotation.key).to.equal('http.status_code');
+          expect(annotations[5].annotation.value).to.equal('404');
+
+          expect(annotations[6].annotation.annotationType).to.equal('ClientRecv');
+
+          expect(annotations[7]).to.be.undefined; // eslint-disable-line no-unused-expressions
+
+          done();
+        });
+      });
+    });
+  });
+
+
+  it('should report when service does not exist', done => {
+    tracer.scoped(() => {
+      api.listen(0, () => {
+        const zipkinRequest = new Request(tracer, serviceName, remoteServiceName);
+        const host = 'bad.invalid.url';
+        const url = `http://${host}`;
+        zipkinRequest.get({url, timeout: 100}).catch(() => {
+          const annotations = record.args.map(args => args[0]);
+          const initialTraceId = annotations[0].traceId.traceId;
+          annotations.forEach(ann => expect(ann.traceId.traceId)
+            .to.equal(initialTraceId).and
+            .to.have.lengthOf(16));
+
+          expect(annotations[0].annotation.annotationType).to.equal('ServiceName');
+          expect(annotations[0].annotation.serviceName).to.equal('weather-app');
+
+          expect(annotations[1].annotation.annotationType).to.equal('Rpc');
+          expect(annotations[1].annotation.name).to.equal('GET');
+
+          expect(annotations[2].annotation.annotationType).to.equal('BinaryAnnotation');
+          expect(annotations[2].annotation.key).to.equal('http.url');
+          expect(annotations[2].annotation.value).to.equal(url);
+
+          expect(annotations[3].annotation.annotationType).to.equal('ClientSend');
+
+          expect(annotations[4].annotation.annotationType).to.equal('ServerAddr');
+
+          expect(annotations[5].annotation.annotationType).to.equal('BinaryAnnotation');
+          expect(annotations[5].annotation.key).to.equal('error');
+          expect(annotations[5].annotation.value)
```

Comment:
I am not entirely sure about this error value. In such cases we add the `error code`. Check https://github.com/openzipkin/zipkin-js/pull/233#issuecomment-403313503

cc @adriancole @chris-smith-zocdoc

Reply:
@jcchavezs I see your point error codes are better than strings. I can remove the assertion but if you want to change the annotation value, it's better to open a new ticket.